# HG changeset patch
# User Rik <octave@nomad.inbox5.com>
# Date 1331500441 25200
#      Sun Mar 11 14:14:01 2012 -0700
# Node ID 2232931f24947fa56225866d2424ec29de1567c9
# Parent  8bfa6e2bb4ed9623a88f7078d323ad4722dc4123
Return regexp output when input is a cellstr and there is no return variable assignment.

regexp.cc (regexp, regexpi): Return output when nargout == 0.

diff --git a/src/DLD-FUNCTIONS/regexp.cc b/src/DLD-FUNCTIONS/regexp.cc
--- a/src/DLD-FUNCTIONS/regexp.cc
+++ b/src/DLD-FUNCTIONS/regexp.cc
@@ -636,17 +636,17 @@ Alternatively, use (?x) in the pattern.\
 {
   octave_value_list retval;
 
   int nargin = args.length ();
 
   if (nargin < 2)
     print_usage ();
   else if (args(0).is_cell () || args(1).is_cell ())
-    retval = octcellregexp (args, nargout, "regexp");
+    retval = octcellregexp (args, (nargout > 0 ? nargout : 1), "regexp");
   else
     retval = octregexp (args, nargout, "regexp");
 
   return retval;
 }
 
 /*
 
@@ -884,17 +884,17 @@ syntax of the search pattern.\n\
 {
   octave_value_list retval;
 
   int nargin = args.length ();
 
   if (nargin < 2)
     print_usage ();
   else if (args(0).is_cell () || args(1).is_cell ())
-    retval = octcellregexp (args, nargout, "regexpi", true);
+    retval = octcellregexp (args, (nargout > 0 ? nargout : 1), "regexpi", true);
   else
     retval = octregexp (args, nargout, "regexpi", true);
 
   return retval;
 }
 
 /*
 
